File: VehicleServer/Entities/Item.cs
using System.ComponentModel.DataAnnotations.Schema;
using VehicleServer.DTOs;
using VehicleServer.Enums;

namespace VehicleServer.Entities
{
    [Table("StockItems")]
    public class Item
    {
        public int ItemId { get; set; }
        public string? Name { get; set; }
        public string? NameAm { get; set; }
        public string? Description { get; set; }
        public int CategoryId { get; set; }
        public Category? Category { get; set; }
        public bool? IsDeleted { get; set; }

        public bool? IsPlate { get; set; }

        public ItemTypeEnum ItemTypeCode { get; set; }

        // New navigation properties
        public ICollection<Stock>? Stock { get; set; }
        public ICollection<StockTransaction>? StockTransactions { get; set; }

    }

}

File: VechicleClient/src/app/Models/item.ts
// import { PlatePool } from "./PlatePool";

import { ItemTypeCodeEnum } from "../enums/item-type-code.enum";

export interface Item {
    itemId: number;
    name: string;
    description: string;
    nameAm: string;
    descriptionAm: string;
    categoryId: number;
    isPlate: boolean;
    itemTypeCode: ItemTypeCodeEnum;
  }